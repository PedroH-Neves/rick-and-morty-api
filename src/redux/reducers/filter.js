import { ADD_FILTER, REMOVE_FILTER } from "../actions/actions";

const filter = (state = [], action) => {
  switch (action.type) {
    case ADD_FILTER:
      return [...state, action.payload];
    case REMOVE_FILTER:
      return [];
    default:
      return state;  
  }
}

export default filter;