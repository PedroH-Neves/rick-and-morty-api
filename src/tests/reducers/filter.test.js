import filter from '../../redux/reducers/filter';

describe('Filter Characters Reducer', () => {
  it('should return initial state', () => {
    expect(filter(undefined, {})).toMatchSnapshot();
  });

  it('should handle add a new filter', () => {
    expect(filter([],
      {
        type: 'ADD_FILTER',
      })).toMatchSnapshot();
  });

  it('should handle remove filters', () => {
    expect(filter([],
      {
        type: 'REMOVE_FILTER',
      })).toMatchSnapshot();
  });
});
