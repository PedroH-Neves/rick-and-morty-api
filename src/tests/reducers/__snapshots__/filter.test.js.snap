// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Characters Reducer should handle add a new filter 1`] = `
Array [
  undefined,
]
`;

exports[`Filter Characters Reducer should handle remove filters 1`] = `Array []`;

exports[`Filter Characters Reducer should return initial state 1`] = `Array []`;
